thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datasource/cache/ResultSetGroupCacheLoader.java

```diff
@@ -102,4 +104,47 @@ private static String format(ResultSetGroup result) {
     }
     return sb.toString();
   }
+
+  private static Connection[] fromHostList(final String[] thirdeyeBrokers) throws Exception {
+    ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
Close the pool at the end of this method. You could use AnomalyUtils.safelyShutdownExecutionService to shut down the pool. We might want to move safelyShutdownExecutionService to some other Utils class if it is used everywhere.